rust/connlib/shared/src/control.rs

```diff
@@ -21,32 +21,16 @@ use tokio_tungstenite::{
     tungstenite::{self, handshake::client::Request},
 };
 use tungstenite::Message;
-use url::Url;
 
 use crate::{get_user_agent, Error, Result};
+use phoenix_channel::LoginUrl;
```

Comment:
Our control plane and portal are both built in Elixir. The portal uses Phoenix LiveView for real time updates of the UI. Phoenix LiveView is built on top of a generic pub-sub system that works with WebSockets.

The `phoenix-channel` Rust crate is our implementation of a client for the same pubsub system. It is used by gateways, relays and soon the clients.

We have two implementations because Gabi wrote one for connlib and I wrote one for the relay.